frame/uniques/src/impl_locker.rs

```diff
@@ -0,0 +1,33 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation for `Locker` trait
+
+/// Trait to handle asset locking mechanism to ensure interactions with the asset can be implemented
+/// downstream to extend logic of Uniques current functionality
+#[allow(clippy::upper_case_acronyms)]
+pub trait Locker<ClassId, InstanceId> {
```

Comment:
This trait should probably be placed in `frame_support::traits`

Reply:
Would it be appropriate for this trait to go into the `misc.rs` file since it is a smaller trait?

Reply:
probably misc is fine, but please do expose it correctly so i can access from `frame_support::traits`

Reply:
I realized I can do this and looks cleaner:
- Remove file `impl_locker.rs` since `pallet_uniques` doesn't need to implement it
- Add the `impl_locker.rs` code to `misc.rs` in `frame_support::traits`
- Expose `Locker` in the `traits.rs` file in `frame_support` 
- Update the `pallet_uniques` `lib.rs` to include `frame_support::traits::Locker`